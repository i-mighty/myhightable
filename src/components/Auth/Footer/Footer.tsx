import React from 'react';
import styled from 'styled-components/native';
import { Footer as NBFooter, Text, Left, Body } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';
import Banner from '@src/assets/highTableLogo.svg';

interface FooterProps {
  testID?: string;
}

const StyledFooter = styled(NBFooter)<FooterProps>`
  background-color: ${({ theme }) => theme.vars.white};
  width: 100%;
  margin-top: 5px;
  elevation: 0;
`;

const Footer: React.FC<FooterProps> = ({ testID, children }) => {
  return <StyledFooter testID={testID}>{children}</StyledFooter>;
};

export default Footer;
