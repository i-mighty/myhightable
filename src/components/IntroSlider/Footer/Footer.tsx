import React from 'react';
import styled from 'styled-components/native';
import { Footer as NBFooter, View } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';

interface ProgressStepProps {
  position: number;
  testID?: string;
}

interface FooterProps {
  currentSlide: number;
  testID?: string;
  stepTestID?: string;
}

const StyledFooter = styled(NBFooter)`
  top: -50px;
  margin-bottom: -60px;
  background-color: transparent;
  elevation: 0;
`;

const ProgressStepper = styled(View)`
  width: 200px;
  height: 4px;
  background-color: ${({ theme }) => colorOpacity(theme.vars.orange, 0.24)};
`;

const ProgressStep = styled(View)<ProgressStepProps>`
  width: 20%;
  height: 0px;
  border: 2px solid ${({ theme }) => theme.vars.orange};
  position: relative;
  left: ${({ position }) => `${position ? position * 20 : '0'}%`};
`;

const Footer: React.FC<FooterProps> = ({
  currentSlide,
  testID,
  stepTestID,
}) => {
  return (
    <StyledFooter>
      <ProgressStepper testID={testID}>
        <ProgressStep position={currentSlide} testID={stepTestID} />
      </ProgressStepper>
    </StyledFooter>
  );
};

export default Footer;
