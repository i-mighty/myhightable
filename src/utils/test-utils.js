import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
