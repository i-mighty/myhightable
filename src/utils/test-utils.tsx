import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
