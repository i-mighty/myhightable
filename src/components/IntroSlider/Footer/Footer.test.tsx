import Footer from './Footer';
import { renderWithProps } from '@src/utils/test-utils';

const defaultProps = {
  currentSlide: 0,
  testID: 'progress',
  stepTestID: 'step',
};
const updatedSlideProps = { ...defaultProps, currentSlide: 1 };
describe('Footer', () => {
  it('renders successfully', () => {
    const { getByTestId, baseElement } = renderWithProps(Footer, defaultProps);
    const progressBar = getByTestId(defaultProps.testID);
    const progressStep = getByTestId(defaultProps.stepTestID);
    expect(progressBar).toBeDefined();
    expect(progressStep).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });
  it('moves the step with the count ', () => {
    const { getByTestId, baseElement } = renderWithProps(
      Footer,
      updatedSlideProps,
    );
    const progressStep = getByTestId(updatedSlideProps.stepTestID);
    expect(progressStep).toHaveStyleRule(
      'left',
      `${updatedSlideProps.currentSlide * 20}%`,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
