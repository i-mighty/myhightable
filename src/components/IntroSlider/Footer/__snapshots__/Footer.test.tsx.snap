// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer moves the step with the count  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderColor": "#cbcbcb",
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 55,
              "justifyContent": "center",
              "left": 0,
              "paddingBottom": 0,
              "right": 0,
            },
            Object {
              "backgroundColor": "transparent",
              "elevation": 0,
              "marginBottom": -60,
              "top": -50,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "rgba(248,149,5,0.24)",
                "height": 4,
                "width": 200,
              },
            ]
          }
          testID="progress"
        >
          <View
            position={1}
            style={
              Array [
                Object {},
                Object {
                  "borderColor": "#F89505",
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "height": 0,
                  "left": "20%",
                  "position": "relative",
                  "width": "20%",
                },
              ]
            }
            testID="step"
          />
        </View>
      </View>
    </SafeAreaView>
  </View>
</View>
`;

exports[`Footer renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderColor": "#cbcbcb",
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 55,
              "justifyContent": "center",
              "left": 0,
              "paddingBottom": 0,
              "right": 0,
            },
            Object {
              "backgroundColor": "transparent",
              "elevation": 0,
              "marginBottom": -60,
              "top": -50,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "backgroundColor": "rgba(248,149,5,0.24)",
                "height": 4,
                "width": 200,
              },
            ]
          }
          testID="progress"
        >
          <View
            position={0}
            style={
              Array [
                Object {},
                Object {
                  "borderColor": "#F89505",
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "height": 0,
                  "left": "0%",
                  "position": "relative",
                  "width": "20%",
                },
              ]
            }
            testID="step"
          />
        </View>
      </View>
    </SafeAreaView>
  </View>
</View>
`;
